\section{Introduction}
Clustering is the task of grouping objects so that, based on a given similarity
threshold, each object belongs to only one cluster.  This project is concerned
with the clustering of DNA and RNA sequences. Our clustering method strives to
find one centroid, a sequence that represents a cluster, that is within some
given distance of a sequence, but the centroid is not guaranteed to be the best
matching centroid.

The main motivation for this project comes from a need for efficient tools for
clustering of sequence data, and related techniques, in the microbiology
department at the University of Copenhagen. In particular, the idea for this
project comes from a collaboration between the supervisor of this project,
Rasmus Fonseca, and Martin Asser Hansen, who is a bioinformatician in the
Molecular Microbial Ecology
Group\footnote{\url{http://www2.bio.ku.dk/microbiology/}}.

Clustering huge amounts of DNA/RNA-sequences (up to 500 million strings of
500-1500 characters each) is computationally hard and there are not many
available algorithms and tools for efficient clustering of sequencing data. The
one tool \texttt{UCLUST}\footnote{\url{http://drive5.com/usearch/}} that does
the job is closed-source and considered too expensive.

This project researches the possibilities of creating an open source tool that
can match the performance of the proprietary version of \texttt{UCLUST}. We
present an implementation of a solution that uses the concept of $k$-mer
counting, for measuring the distance between sequences, and utilizes the
$k$-mers occuring in the centroids for improving the search for a centroid that
is within the given threshold distance.
