\documentclass[11pt,a4paper]{article}

\usepackage{tikz}
\usetikzlibrary{shapes}
\usepackage[utf8]{inputenc}
\usepackage{amsmath}
\usepackage{mathtools}
\usepackage{amsfonts}
\usepackage{pdfpages}
\usepackage{gauss}
\usepackage{fancyvrb}
\usepackage{hyperref}
\usepackage{graphicx}
\usepackage{url}
\usepackage{float}
\usepackage[bottom]{footmisc}

% headers and footers
\usepackage{fancyhdr, lastpage}
\pagestyle{fancy}
\fancyhf{}
\renewcommand{\headrulewidth}{0pt}
\cfoot{Page \thepage\ of \pageref{LastPage}}

\title{Bachelor project \\
       \vspace{2mm}
       {\LARGE Efficient DNA/RNA-sequence clustering}}
\author{Anders Kiel Hovgaard \and Nikolaj Dybdahl Rathcke}

\begin{document}
\maketitle
\thispagestyle{fancy}

\begin{abstract}
  This will eventually contain a beautiful abstract...
\end{abstract}

\section{What should be in report?}

\begin{itemize}
  \item Introduction: introducing the motivation and goals for the project.
    
  \item Terminology: clarification of USEARCH vs UCLUST, definition of
    clustering and distance metric etc.

  \item Biology: the basics about gene sequences, DNA, RNA, sequencing etc.

  \item Background: theoretical overview of the field of clustering
    \begin{itemize}
      \item Types of cluster analysis algorithms
        \begin{itemize}
          \item Hierarchical clustering (agglomerative and divisive)
          \item Graph based clustering
          \item Gready algorithm like in \texttt{UCLUST}
          \item Gready algorithm with recalculation of centroids
        \end{itemize}

      \item Distance metrics
        \begin{itemize}
          \item Edit distance (Levenshtein)
          \item d2 distance (feature based distance)
          \item Sequence alignment
        \end{itemize}

      \item Clustering "history"
    \end{itemize}

  \item Research process
    \begin{itemize}
      \item Implementing basic Levenshtein and memoized Levenshtein,
        implementing d2 distance and comparing these.
      \item Testing USEARCH 32-bit on real data
      \item Implementing very (almost naively) gready clustering algorithm.
      \item Testing clustering algorithm with d2 distance and comparing
        performance to USEARCH.
    \end{itemize}
\end{itemize}


\section{To be done}
d2 - Remembering/reading kmers somehow instead of recomputing each time?


\section{Background}

\subsection{Distance metrics}

\subsubsection{Edit distance}
One type of \emph{edit distance} is the \emph{Levenshtein} distance, which is a string metric for determining the similarity between two sequences. It is defined to be the minimum number of edits to turn the first sequence into the other. \\
The edit operations consists of \emph{insertions}, \emph{deletions} and \emph{substitutions}. These operations are, respectively, inserting a letter, removing a letter and changing one letter for another. \\
For example, the two sequences
\begin{center}
\texttt{ACGT} \\
\texttt{ACGGC}
\end{center}
would have a distance of 2 (substituting T for G and inserting a C). \\
However, there are some cases where the relevance of the distance is arguable. Consider the sequences
\begin{center}
\texttt{AACC} \\
\texttt{CCAA}
\end{center}
with a distance of 4. The two sequences actually have the maximal distance possible even though they have a close resemblance to each other.

\subsubsection{Feature based distance}



\subsubsection{Sequence alignment}
\emph{Sequence alignment} is used in bioinformatics to identify regions of similarity by arranging two or more sequences in a certain manner. Besides shifting a sequence to a side, gaps can be inserted between objects as well. We represent the gap with '-' and it indicates a insertion in the sequence or a deletion from another sequence.\\
We call a column that includes - for an indel. If it does not and all object in the column are same, then we have a match. Otherwise, it is a mismatch.\\
\begin{figure}[h!]
  \centering
  	  \begin{align*}
  	  	\mbox{\texttt{ATGCA}} \\
      	\mbox{\texttt{-TGCG}}
  	  \end{align*}
  \caption{Sequence alignment of the sequences 'ATGCA' and 'TGCG'}
  \label{fig:seqAlignment}
\end{figure}
Figure \ref{fig:seqAlignment} displays an alignment of two sequences. Column 1 is an indel, column 2-4 are matches and column 5 is a mismatch. The amount of each type can then be used in a formula to calculate the similarity between the sequences. Note that more than two sequences can be aligned in a \emph{multiple sequence alignment}. \\
There are many alignments. Among those is an optimal one which is what is strived to be found.


\begin{thebibliography}{9}
  \bibitem[Hazelhurst2003]{hazelhurst2003}
    Scott Hazelhurst,
    \emph{An implementation of the $d^2$ distance function for DNA
      sequences: The wcd $d^2$ EST clustering algorithm},
      September 2003,
      \url{http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.9.4289&rep=rep1&type=pdf}.
\end{thebibliography}

\end{document}
