\documentclass[11pt,a4paper]{article}

\usepackage{tikz}
\usetikzlibrary{shapes}
\usepackage[utf8]{inputenc}
\usepackage{amsmath}
\usepackage{mathtools}
\usepackage{amsfonts}
\usepackage{pdfpages}
\usepackage{gauss}
\usepackage{fancyvrb}
\usepackage{hyperref}
\usepackage{graphicx}
\usepackage{url}
\usepackage{float}
\usepackage[bottom]{footmisc}

% algorithms
\usepackage{algorithm}              % algorithm float environment
\usepackage[noend]{algpseudocode}   % from algorithmicx; no "end" on functions
\newcommand*\Let[2]{\State #1 $\gets$ #2}
\algrenewcommand\algorithmicrequire{\textbf{Precondition:}}
%\algrenewcommand\alglinenumber[1]{\\footnotesize{\color{gray}#1}}

% headers and footers
\usepackage{fancyhdr, lastpage}
\pagestyle{fancy}
\fancyhf{}
\renewcommand{\headrulewidth}{0pt}
\cfoot{Page \thepage\ of \pageref{LastPage}}

% handy commands etc.
\newcommand{\eqdef}{\overset{\mathrm{def}}{=\joinrel=}}

\title{Bachelor project: Midway report \\
       \vspace{2mm}
       {\LARGE Efficient DNA/RNA-sequence clustering}}
\author{Anders Kiel Hovgaard \and Nikolaj Dybdahl Rathcke}

\begin{document}
\maketitle
\thispagestyle{empty}
\newpage

\begin{abstract}
  This will eventually contain a beautiful abstract...
\end{abstract}
\thispagestyle{plain}
\pagenumbering{roman}
\newpage

\tableofcontents
\thispagestyle{plain}
\newpage

\thispagestyle{fancy}
\pagenumbering{arabic}
\section{What should be in report?}

\begin{itemize}
  \item Introduction: introducing the motivation and goals for the project.
    
  \item Terminology: clarification of USEARCH vs UCLUST, definition of
    clustering and distance metric etc.

  \item Biology: the basics about gene sequences, DNA, RNA, sequencing etc.

  \item Background: theoretical overview of the field of clustering
    \begin{itemize}
      \item Types of cluster analysis algorithms
        \begin{itemize}
          \item Hierarchical clustering (agglomerative and divisive)
          \item Graph based clustering
          \item Gready algorithm like in \texttt{UCLUST}
          \item Gready algorithm with recalculation of centroids
        \end{itemize}

      \item Distance metrics
        \begin{itemize}
          \item Edit distance (Levenshtein)
          \item d2 distance (feature based distance)
          \item Sequence alignment
        \end{itemize}

      \item Clustering "history"
    \end{itemize}

  \item Research process
    \begin{itemize}
      \item Implementing basic Levenshtein and memoized Levenshtein,
        implementing d2 distance and comparing these.
      \item Testing USEARCH 32-bit on real data
      \item Implementing very (almost naively) gready clustering algorithm.
      \item Testing clustering algorithm with d2 distance and comparing
        performance to USEARCH.
    \end{itemize}
\end{itemize}

\section{To be done}

\newpage


\input{introduction.tex}
\input{terminology.tex}
\input{background.tex}
\input{research_process.tex}

\newpage
\nocite{dong, hazelhurst, usearch, uclust}

\bibliography{report}
\bibliographystyle{alpha} % plain, abbrv


\end{document}
