\documentclass[11pt,a4paper]{article}

\usepackage[dot, autosize, outputdir="dotgraphs/"]{dot2texi}
\usepackage{tikz}
\usetikzlibrary{shapes}
\usepackage[utf8]{inputenc}
\usepackage{amsmath}
\usepackage{mathtools}
\usepackage{amsfonts}
\usepackage{pdfpages}
\usepackage{gauss}
\usepackage{fancyvrb}
\usepackage{graphicx}
\usepackage{url}
\usepackage{float}
\usepackage[bottom]{footmisc}

% headers and footers
\usepackage{fancyhdr, lastpage}
\pagestyle{fancy}
\fancyhf{}
\renewcommand{\headrulewidth}{0pt}
\cfoot{Page \thepage\ of \pageref{LastPage}}

% Gantt diagrams
\usepackage{gantt}

% Alternative titles:
%   "Efficient clustering of massive sequencing data"
%   "Efficient DNA/RNA-sequence clustering"
%   "Efficient clustering of massive amounts of DNA/RNA-sequences"
\title{Bachelor project: Synopsis \\
       \vspace{2mm}
       {\LARGE Efficient DNA/RNA-sequence clustering}}
\author{Anders Kiel Hovgaard \and Nikolaj Dybdahl Rathcke}
\date{February 23, 2015}

\begin{document}
\maketitle
\thispagestyle{fancy}

\section{Problem statement}
Can we implement an algorithm that can cluster DNA/RNA-sequencing data, of at
least half a million sequences of 500-1500 characters each, and compete with
the performance of \texttt{UCLUST}? If not, how close can we get?


\subsection{Limitations on the scope of the project}
The algorithm might either be one that we design from scratch or a modified
version of an existing algorithm. Similarly for the implementation, we might
reuse parts of existing open implementations to some extent.

Clustering of protein sequences will not be considered in this project,
i.e. only RNA and DNA sequences will be considered.

The implementation will be developed on GNU/Linux systems and will not be
guaranteed to work on other operating systems.

We will only write a brief user manual for the program.


\section{Motivation}
The main motivation for this project comes from a need for efficient tools for
clustering of sequence data, and related techniques, in the microbiology
department at the University of Copenhagen. In particular, the idea for this
project comes from a collaboration between the supervisor of this project,
Rasmus Fonseca, and Martin Asser Hansen, who is a bioinformatician in the
Molecular Microbial Ecology
Group\footnote{\url{http://www2.bio.ku.dk/microbiology/}}.

The problem consists of clustering huge amounts of DNA/RNA-sequences (up to 500
million strings of 500-1500 characters each), based on a given similarity
threshold, such that each sequence belongs to only one cluster and the
sequences within a cluster are more similar to each other than to sequences in
other clusters.
 
There are not many available algorithms and tools for efficient clustering of
sequencing data. The one tool
\texttt{UCLUST}\footnote{\url{http://drive5.com/usearch/}} that does the job is
closed-source and considered too expensive.

The goal of this project is to research the possibilities of creating an open source tool that can match the performance of the proprietary version of
\texttt{UCLUST}.


\section{Tasks and time planning}
The following tasks are not necessarily dependent on each other in the order
presented, for instance, the implementation of the distance metric and
clustering algorithm will be interleaved with testing and optimization.
Thus, some of the tasks will be completed in parallel. Furthermore, the report
will be written concurrently with some of the tasks.

A Gantt diagram for the project plan is shown in figure \ref{fig:gantt}.

\begin{itemize}
  \item Survey existing literature, algorithms and implementations of sequence
    clustering. ($\sim$1-2 weeks\footnote{Approximate workload, not necessarily
    actual duration of task.})

  \item Choose distance metric and parameters, design algorithm for calculating
    distance. ($\sim$1 week)

  \item Design initial version of clustering algorithm. ($\sim$1-2 weeks)

  \item Implement distance metric. ($\sim$1-2 weeks)

  \item Implement clustering algorithm. ($\sim$3-4 weeks)

  \item Analyze and profile the implementation, investigate possibilities for
    optimization and perform these on the implementation. Possibly perform
    parallelization of the implementation. ($\sim$3-4 weeks)

  \item Perform ongoing testing after implementation of initial algorithm and
    in-between optimizations and modifications. ($\sim$3-4 weeks)

  \item Analyze the time and space complexity of the algorithm. ($\sim$1-2
    weeks)
\end{itemize}

\begin{figure}[H]
  \begin{gantt}[xunitlength=0.5cm,fontsize=\small,titlefontsize=\small]{12}{20}
    \begin{ganttitle}
      \titleelement{Feb}{4}
      \titleelement{Mar}{4}
      \titleelement{Apr}{4}
      \titleelement{May}{4}
      \titleelement{Jun}{4}
    \end{ganttitle}
    \ganttbar{Alg. design}{2}{4}
    \ganttbar{Impl. of metric}{4}{7}
    \ganttbar{Impl. of alg.}{4}{7}
    \ganttmilestone[color=green]{Prototype}{8}
    \ganttbar{Optimization}{8}{7}
    \ganttbar{Testing}{5}{10}
    \ganttbar{Complexity}{12}{3}
    \ganttbar{Writing}{4}{13}
    \ganttmilestone[color=cyan]{Mid-way report}{10}
    \ganttmilestone[color=cyan]{Report}{17}
    \ganttmilestone[color=red]{Defense}{18}
  \end{gantt}
  \caption{Gantt diagram of the project plan.}
  \label{fig:gantt}
\end{figure}


\section{Product}
We will design and implement an algorithm for efficient clustering of massive
amounts of DNA/RNA-sequences, most likely in a programming language such as
\texttt{C} or \texttt{C++}. Furthermore, we will write a report that documents
our efforts, the algorithms and implementations produced and evaluates their
performance.


\end{document}
